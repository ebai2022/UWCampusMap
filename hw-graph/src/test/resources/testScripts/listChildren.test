# Test list children

# Test empty case aka island
CreateGraph A
AddNode A e1
ListChildren A e1

# Test single case
AddNode A e2
AddEdge A e1 e2 test
ListChildren A e1

# Test multi case
AddNode A e3
AddNode A e4
AddEdge A e3 e4 hello
AddEdge A e4 e3 backwards
AddEdge A e3 e1 multiple
ListChildren A e1
ListChildren A e2
ListChildren A e3
ListChildren A e4