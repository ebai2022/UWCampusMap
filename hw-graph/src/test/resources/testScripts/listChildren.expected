# Test list children

# Test empty case aka island
created graph A
added node e1 to A
the children of e1 in A are:

# Test single case
added node e2 to A
added edge test from e1 to e2 in A
the children of e1 in A are: e2(test)

# Test multi case
added node e3 to A
added node e4 to A
added edge hello from e3 to e4 in A
added edge backwards from e4 to e3 in A
added edge multiple from e3 to e1 in A
the children of e1 in A are:
the children of e2 in A are:
the children of e3 in A are:
the children of e4 in A are: